client/src/pages/MapPage/MapViewer.vue

```diff
@@ -84,6 +65,43 @@ export default {
   },
 
   methods: {
+    /** Adds the map and control elements to the page, ready for a data source*/
+    initialiseMapCanvas() {
+      this.map = new MapView({
+        canvas: this.$refs.map,
+        theme: this.customisedTheme,
+      });
+
+      // Add control buttons to map
+      const controls = new MapControls(this.map);
+      const ui = new MapControlsUI(controls, {projectionSwitch: true});
+      this.$refs.map.parentElement.appendChild(ui.domElement);
+
+      // Make map fullscreen
+      this.map.resize(window.innerWidth, window.innerHeight);
+
+      // Make map sizing responsive to window size
+      window.addEventListener("resize", () => {
+        this.map.resize(window.innerWidth, window.innerHeight);
+      });
+    },
+
+    /** Adds terrain, street, building, etc. data sources to the map */
+    initaliseBaseDataSources() {
+      // Retrieve tiled base map
+      if (this.token) {
+        const omvDataSource = new OmvDataSource({
+          baseUrl: "https://xyz.api.here.com/tiles/herebase.02",
```

Comment:
is baseUrl a hard coded value? I wonder if it could benifit from being a CONST class member? You do have it here defined within a const variable - so perhaps it's not nessecary. I guess it depends if you expect to reference it elsewhere (the baseUrl not the omvDataSource), or if you are likely to need to update it in the future. If either of those are the case I think a class member may be better. Your decision :)

Reply:
We would like to swap to other data sources other than here.com in the future I'm sure but I'm going to add that as a new issue #15 and leave it hard-coded for now. I will need a way of swapping between data sources and style sets dynamically without reloading the whole component so it's going to need more work than just pulling it out to a class member. Thanks for the idea, but I will put it in the backlog for now.

Reply:
Sweet - good the see you're already thing about it - and yes a const class member isn't appropriate if you're looking at changing dynamically!